server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java

```diff
@@ -407,8 +407,11 @@ public CompletableFuture<Revision> updatePerRolePermissions(Author author,
         requireNonNull(perRolePermissions, "perRolePermissions");
 
         if (Project.isReservedRepoName(repoName)) {
-            throw new UnsupportedOperationException(
-                    "can't update the per role permission for internal repository: " + repoName);
+            final Set<Permission> guest = perRolePermissions.guest();
+            if (!guest.isEmpty()) {
+                throw new UnsupportedOperationException(
+                        "can't give a permission to guest for internal repository: " + repoName);
```

Comment:
For consistency with the permission of `internalPermissions`, should we disallow giving permissions to `MEMBER` for internal repositories?
https://github.com/line/centraldogma/blob/0cf273402d70773e0aab1e2b858111a1c1c0f22c/server/src/main/java/com/linecorp/centraldogma/server/metadata/PerRolePermissions.java#L55-L56

Reply:
I believed we had previously discussed this matter and reached a consensus to delegate permission-granting authority to the owner for members. 😉 
`internalPermissions` is assigned to internal repositories upon creation, and the owner has the discretion to change it at will. While I'm not strongly opposed, I see no compelling reason to restrict granting permissions to members.

Reply:
Understood.
I was confused that the permission could be also granted to `dogma` repo.

Reply:
Technically, I believe allowing MEMBER access to `dogma` can potentially allow MEMBERs to "kick" OWNERs out of projects.
I don't really know the distinction of owner and member though. If the meaning is simply to add an extra layer of authority, I guess the current change is fine.

Reply:
Ah, got it. The dogma repo is restricted from being accessed by any user, so I initially thought this would suffice: https://github.com/line/centraldogma/blob/0cf273402d70773e0aab1e2b858111a1c1c0f22c/server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/RequiresPermissionDecorator.java#L78-L83.

However, for clarity, I believe it's beneficial to explicitly allow changes only to the meta repository.